component/core/src/main/java/org/exoplatform/social/core/metadata/attachment/AttachmentServiceImpl.java

```diff
@@ -128,23 +128,72 @@ public ObjectAttachmentOperationReport createAttachments(ObjectUploadResourceLis
           + " doesn't have enough permissions to attach files on object " + objectType + "/" + objectId);
     }
 
+    List<String> uploadIds = attachmentList.getUploadIds();
+    if (CollectionUtils.isEmpty(uploadIds)) {
+      throw new IllegalArgumentException("Attachment uploadIds is mandatory");
+    }
+    return attachUploadFiles(uploadIds, objectType, objectId, parentObjectId, userIdentityId);
+  }
+
+  @Override
+  public ObjectAttachmentOperationReport updateAttachments(FileAttachmentResourceList attachmentList,
+                                                           Identity userAclIdentity) throws ObjectNotFoundException,
+                                                                                     IllegalArgumentException,
+                                                                                     IllegalAccessException {
+
+    if (attachmentList == null) {
+      throw new IllegalArgumentException("Attachment is mandatory");
+    }
+
+    long userIdentityId = attachmentList.getUserIdentityId();
+    if (userIdentityId <= 0) {
+      throw new IllegalArgumentException("User identity id is mandatory");
+    }
+
+    org.exoplatform.social.core.identity.model.Identity userIdentity =
+                                                                     identityManager.getOrCreateUserIdentity(userAclIdentity.getUserId());
+
+    if (userIdentity == null || userIdentity.isDeleted() || !userIdentity.isEnable()) {
+      throw new IllegalStateException("User with id " + userAclIdentity.getUserId() + " isn't valid");
+    }
+    String objectType = attachmentList.getObjectType();
+    if (StringUtils.isBlank(objectType)) {
+      throw new IllegalArgumentException("Object type is mandatory");
+    }
+    String objectId = attachmentList.getObjectId();
+    if (StringUtils.isBlank(objectId)) {
+      throw new IllegalArgumentException("Object identifier is mandatory");
+    }
+
+    MetadataObject metadataObject = new MetadataObject(objectType, objectId);
+
+    List<MetadataItem> existingAttachments =
+                                           metadataService.getMetadataItemsByMetadataTypeAndObject(AttachmentService.METADATA_TYPE.getName(),
+                                                                                                   metadataObject);
+    if (existingAttachments.isEmpty()) {
```

Comment:
You mean when not empty, and please add unit test for this failing Test case

```suggestion
    if (!existingAttachments.isEmpty()) {
```

Reply:
yes